app/src/main/java/chat/rocket/android/renderer/MessageRenderer.java

```diff
@@ -35,7 +35,7 @@ public MessageRenderer avatarInto(ImageView imageView, String hostname) {
         imageView.setImageResource(R.drawable.ic_error_outline_black_24dp);
         break;
       default:
-        userRenderer.avatarInto(imageView, hostname);
+        userRenderer.avatarInto(imageView, hostname, object.getAvatar());
```

Comment:
I think UserRenderer should not know Message's avatar.
It would be better to check the presence of Message's avatar here, and request to UserRenderer to render avatar image only when Message's avatar is absent.

```
if (object.getAvatar != null) {
  loadImageInto(object.getAvatar(), imageView);  // just load image with Picasso.
} else {
  userRenderer.avatarInto(imageView, hostname);
}
```